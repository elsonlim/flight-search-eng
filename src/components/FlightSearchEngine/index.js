import React, { Component } from 'react';
import { Grid } from 'semantic-ui-react';

import FlightList from '../FlightList';
import SearchContainer from '../SearchContainer';
import './FlightSearchEngine.css';

class FlightSearchEngine extends Component {
    render () {
        return (
            <Grid stackable>
                <Grid.Row>
                <Grid.Column width={4}>
                    <SearchContainer />
                </Grid.Column>

                <Grid.Column width={12}>
                    <FlightList />
                </Grid.Column>
                </Grid.Row>
            </Grid>
        );
    }
}

export default FlightSearchEngine;