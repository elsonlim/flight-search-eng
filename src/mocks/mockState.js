import moment from 'moment';

export const flights = [
    {
        "id": 0,
        "IATA": "DL",
        "flightNo": 90,
        "from": "USA",
        "to": "HKG",
        "price": {"amount": 261, "currency": "USD"},
        "time": {"arrival": 609, "departure": 425, "duration": 104}
    }, {
        "id": 1,
        "IATA": "DL",
        "flightNo": 91,
        "from": "HKG",
        "to": "USA",
        "price": {"amount": 232, "currency": "USD"},
        "time": {"arrival": 1956, "departure": 1730, "duration": 146}
    },
];

export const searchParams = {
    isReturn: true,
    from: 'USA',
    to: 'HKG',
    depatureDate: moment(new Date()),
    arrivalDate: moment(new Date()),
    numberOfPassengers: 1,
}

export const priceRange = {
    filter: {min: 0, max: 2000},
    search: {min: 200, max: 1000},
}

export default {
    flights,
    searchType: false,
    searchParams,
    priceRange
};

    