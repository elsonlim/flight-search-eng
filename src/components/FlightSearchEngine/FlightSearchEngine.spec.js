import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import mockState from '../../mocks/mockState';
import FlightSearchEngine from './index';

describe('FlightSearchEngine', () => {
    let props;
    let wrapper;
    let store;
    const mockStore = configureStore();

    beforeEach(() => {
        store =  mockStore(mockState);
    });

    describe('snapshot test', () => {
		test('show airline logo', () => {
			const tree = renderer.create(<Provider store={store} ><FlightSearchEngine /></Provider>).toJSON();
			expect(tree).toMatchSnapshot();
		});
	});
});
