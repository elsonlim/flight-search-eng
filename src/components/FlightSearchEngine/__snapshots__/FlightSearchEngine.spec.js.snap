// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightSearchEngine snapshot test show airline logo 1`] = `
<div
  className="ui stackable grid"
>
  <div
    className="row"
  >
    <div
      className="four wide column"
    >
      <div>
        <div
          className="ui tabular menu"
        >
          <a
            className="active item"
            onClick={[Function]}
          >
            One Way
          </a>
          <a
            className="item"
            onClick={[Function]}
          >
            Return
          </a>
        </div>
        <div>
          <form
            action={undefined}
            className="ui form"
            onSubmit={[Function]}
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            <div>
              From
            </div>
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              className="ui fluid search selection dropdown form-field"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={undefined}
            >
              <input
                aria-autocomplete="list"
                autoComplete="off"
                className="search"
                onChange={[Function]}
                style={
                  Object {
                    "width": undefined,
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                className="default text"
                role="alert"
              >
                Select 
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                aria-multiselectable={undefined}
                className="menu transition"
                role="listbox"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="au flag"
                  />
                  <span
                    className="text"
                  >
                    Australia
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="jp flag"
                  />
                  <span
                    className="text"
                  >
                    Japan
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="hk flag"
                  />
                  <span
                    className="text"
                  >
                    Hong Kong
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="sg flag"
                  />
                  <span
                    className="text"
                  >
                    Singapore
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="tw flag"
                  />
                  <span
                    className="text"
                  >
                    Taiwan
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="us flag"
                  />
                  <span
                    className="text"
                  >
                    United States
                  </span>
                </div>
              </div>
            </div>
            <div>
              Destination
            </div>
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              className="ui fluid search selection dropdown form-field"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={undefined}
            >
              <input
                aria-autocomplete="list"
                autoComplete="off"
                className="search"
                onChange={[Function]}
                style={
                  Object {
                    "width": undefined,
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                className="default text"
                role="alert"
              >
                Destination 
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                aria-multiselectable={undefined}
                className="menu transition"
                role="listbox"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="au flag"
                  />
                  <span
                    className="text"
                  >
                    Australia
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="jp flag"
                  />
                  <span
                    className="text"
                  >
                    Japan
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="hk flag"
                  />
                  <span
                    className="text"
                  >
                    Hong Kong
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="sg flag"
                  />
                  <span
                    className="text"
                  >
                    Singapore
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="tw flag"
                  />
                  <span
                    className="text"
                  >
                    Taiwan
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <i
                    className="us flag"
                  />
                  <span
                    className="text"
                  >
                    United States
                  </span>
                </div>
              </div>
            </div>
            <div
              className="form-field"
            >
              <div>
                Depature Date
              </div>
              <div
                className="no-border"
              >
                <div
                  className="SingleDatePicker SingleDatePicker_1 SingleDatePicker__block SingleDatePicker__block_2"
                >
                  <div>
                    <div
                      className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2 SingleDatePickerInput__block SingleDatePickerInput__block_3"
                    >
                      <div
                        className="DateInput DateInput_1 DateInput__block DateInput__block_2"
                      >
                        <input
                          aria-describedby="DateInput__screen-reader-message-date_input"
                          aria-label="Date"
                          autoComplete="off"
                          className="DateInput_input DateInput_input_1"
                          disabled={false}
                          id="date_input"
                          name="date_input"
                          onChange={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder="Date"
                          readOnly={false}
                          required={false}
                          type="text"
                          value="May 20, 2018"
                        />
                        <p
                          className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                          id="DateInput__screen-reader-message-date_input"
                        >
                          Press the down arrow key to interact with the calendar and
                            select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div>
              Passengers
            </div>
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              className="ui fluid search selection dropdown form-field"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={undefined}
            >
              <input
                aria-autocomplete="list"
                autoComplete="off"
                className="search"
                onChange={[Function]}
                style={
                  Object {
                    "width": undefined,
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                className="text"
                role="alert"
              >
                1
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                aria-multiselectable={undefined}
                className="menu transition"
                role="listbox"
              >
                <div
                  aria-checked={true}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="active selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    1
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    2
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    3
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    4
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    5
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    6
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    7
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    8
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    9
                  </span>
                </div>
              </div>
            </div>
            <div
              style={
                Object {
                  "margin": "15px 0",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <button
                className="ui primary button form-button"
                disabled={undefined}
                onClick={[Function]}
                role="button"
                tabIndex={undefined}
                type="submit"
              >
                Search
              </button>
            </div>
          </form>
        </div>
        <div
          className="ui fluid card"
          href={undefined}
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="left aligned header"
            >
              Refine Flight Search
            </div>
            <div
              className="refine-search-slider"
            >
              <div
                style={
                  Object {
                    "marginBottom": "10px",
                    "textAlign": "left",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                      "flexWrap": "wrap",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "fontWeight": "bold",
                        "marginRight": "10px",
                      }
                    }
                  >
                    Price/Pax
                  </span>
                  <span
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#1678C2",
                        "border": "1px solid #e9e9e9",
                        "borderRadius": "5px",
                        "color": "white",
                        "display": "flex",
                        "fontSize": "0.8em",
                        "fontWeight": "normal",
                        "padding": "0 2px",
                        "whiteSpace": "nowrap",
                      }
                    }
                  >
                    200
                     - 
                    1000
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "padding": "5px",
                    }
                  }
                >
                  <div
                    className="rc-slider"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={undefined}
                  >
                    <div
                      className="rc-slider-rail"
                      style={
                        Object {
                          "backgroundColor": "grey",
                        }
                      }
                    />
                    <div
                      className="rc-slider-track rc-slider-track-1"
                      style={
                        Object {
                          "backgroundColor": "#1678C2",
                          "left": "10%",
                          "width": "40%",
                        }
                      }
                    />
                    <div
                      className="rc-slider-step"
                    />
                    <div
                      aria-disabled={false}
                      aria-valuemax={2000}
                      aria-valuemin={0}
                      aria-valuenow={200}
                      className="rc-slider-handle rc-slider-handle-1"
                      onBlur={undefined}
                      onClick={undefined}
                      onContextMenu={undefined}
                      onFocus={undefined}
                      onMouseDown={undefined}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={undefined}
                      role="slider"
                      style={
                        Object {
                          "borderColor": "#1678C2",
                          "left": "10%",
                        }
                      }
                      tabIndex={0}
                    />
                    <div
                      aria-disabled={false}
                      aria-valuemax={2000}
                      aria-valuemin={0}
                      aria-valuenow={1000}
                      className="rc-slider-handle rc-slider-handle-2"
                      onBlur={undefined}
                      onClick={undefined}
                      onContextMenu={undefined}
                      onFocus={undefined}
                      onMouseDown={undefined}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onTouchStart={undefined}
                      role="slider"
                      style={
                        Object {
                          "borderColor": "#1678C2",
                          "left": "50%",
                        }
                      }
                      tabIndex={0}
                    />
                    <div
                      className="rc-slider-mark"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="twelve wide column"
    >
      <div>
        <div
          className="results-header-container"
        >
          <div
            className="results-title"
          >
            Your Results
          </div>
          <div
            className="results-date"
          >
            <div>
              Depart: 
              2018-5-20
            </div>
            <div>
              Return: 
              2018-5-20
            </div>
          </div>
        </div>
        <div
          className="flights"
        >
          <div
            className="ui grid"
          >
            <div
              className="middle aligned row"
            >
              <div
                className="ten wide column"
              >
                <div>
                  <div
                    className="ui equal width grid"
                  >
                    <div
                      className="column"
                    >
                      <div
                        className="flight-details"
                      >
                        <div
                          className="logo"
                        >
                          <div>
                            <img
                              className="ui tiny rounded image"
                              src="/airlineLogo/dl.svg"
                            />
                          </div>
                        </div>
                        <div
                          className="details"
                        >
                          <div>
                            DL
                            -
                            90
                          </div>
                          <div>
                            USA
                             &gt; 
                            HKG
                          </div>
                          <div>
                            Depart: 
                            04:25am
                          </div>
                          <div>
                            Arrival: 
                            06:09am
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      className="column"
                    >
                      <div
                        className="flight-details"
                      >
                        <div
                          className="logo"
                        >
                          <div>
                            <img
                              className="ui tiny rounded image"
                              src="/airlineLogo/dl.svg"
                            />
                          </div>
                        </div>
                        <div
                          className="details"
                        >
                          <div>
                            DL
                            -
                            91
                          </div>
                          <div>
                            HKG
                             &gt; 
                            USA
                          </div>
                          <div>
                            Depart: 
                            05:30pm
                          </div>
                          <div>
                            Arrival: 
                            07:56pm
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="two wide column"
              >
                $
                493
              </div>
              <div
                className="four wide column"
              >
                <button
                  className="ui primary button"
                  disabled={undefined}
                  onClick={[Function]}
                  role="button"
                  tabIndex={undefined}
                >
                  Select flight
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
